src/internal_modules/roc_core/target_openssl/roc_core/secure_random.cpp

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright (c) 2025 Roc Streaming authors
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+#include "secure_random.h"
+
+#include "roc_core/log.h"
+
+#include <openssl/err.h>
+#include <openssl/rand.h>
+
+namespace roc {
+namespace core {
+
+bool secure_random(void* buf, size_t bufsz) {
+    // RAND_priv_bytes() is not needed right now as we do not use these random numbers
+    // privately. See also https://docs.openssl.org/3.0/man7/RAND/
+
+    int ok = RAND_bytes((unsigned char*)buf, (int)bufsz);
+
+    if (ok != 1) {
+        unsigned long err = ERR_get_error();
+        const size_t err_str_len = 256; // minimum buf length is 256
+        char err_str[err_str_len];
+        ERR_error_string_n(err, err_str, err_str_len);
+        roc_log(LogError, "OpenSSL: failed to generate random number: %s", err_str);
+        return false;
+    }
+    return true;
+}
+
+bool secure_random_range_32(uint32_t from, uint32_t to, uint32_t& dest) {
+    // validation `to >= from` exists in the 64-bit version called below
+
+    if (from == 0 && to == UINT32_MAX) {
+        return secure_random(&dest, sizeof(dest));
+    }
```

Comment:
I think this check is not needed in 32 bit version

Reply:
it eliminates mask operations and the loop in line 81